modules/code-editor/src/views/full/store/editor.ts

```diff
@@ -50,6 +53,11 @@ class EditorStore {
 
   @action.bound
   async openFile(file: EditableFile) {
+    if (NO_EDIT_EXTENSIONS.includes(path.extname(file.location))) {
```

Comment:
Why not throw an error and catch on caller side and display a ui warning

Reply:
Lazyness :P I've added a toast. The check is only frontend, it's just to prevent selecting a tgz file and seeing a bunch of hieroglyph